hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentRegistry.java

```diff
@@ -101,6 +94,38 @@ public SubscriptionAssignmentView createSnapshot() {
         return new SubscriptionAssignmentView(snapshot);
     }
 
+    private List<SubscriptionAssignment> readExistingAssignments() {
+        return subscriptionsCache.listActiveSubscriptionNames().stream().flatMap(subscriptionName -> {
+            String path = pathSerializer.serialize(subscriptionName);
+            List<String> children = new ArrayList<>();
+            try {
+                children.addAll(curator.getChildren().forPath(path));
+            } catch (Exception e) {
+                // ignore - should be fixed by the cache
+            }
+            return children.stream().map(node -> new SubscriptionAssignment(node, subscriptionName));
+        }).collect(Collectors.toList());
+    }
+
+    private void onAssignmentAdded(SubscriptionAssignment assignment) {
```

Comment:
this will be called for each assignment twice on startup:

* while warming up the cache
* when events from ZK start to come in

I think we should protect against this kind of obvious duplicates.